core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -389,16 +398,46 @@ protected Node visitShowColumns(ShowColumns showColumns, Void context)
             if (!metadata.schemaExists(session, new CatalogSchemaName(tableName.getCatalogName(), tableName.getSchemaName()))) {
                 throw semanticException(SCHEMA_NOT_FOUND, showColumns, "Schema '%s' does not exist", tableName.getSchemaName());
             }
-            if (metadata.getView(session, tableName).isEmpty() &&
-                    metadata.getTableHandle(session, tableName).isEmpty()) {
+
+            Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
+            if (metadata.getView(session, tableName).isEmpty() && tableHandle.isEmpty()) {
                 throw semanticException(TABLE_NOT_FOUND, showColumns, "Table '%s' does not exist", tableName);
             }
 
-            accessControl.checkCanShowColumns(session.toSecurityContext(), tableName.asCatalogSchemaTableName());
```

Comment:
I think i already somewhat forgot what `checkCanShowColumns` was not sufficient. 
Thanks for capturing the reasoning in the code comment (`// Queries against information_schema should never fail...`). Do you think this can be further  exemplified?

Reply:
I added an unit test to show the example of the problem.